From b013c652ef3889d409c1ce27d632732154a270c0 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 3 Feb 2016 15:38:57 +0200
Subject: [PATCH 0813/1240] clock: a3700: Add SoC bringup mode in 800/800
 preset

- Create anew ddr-800.txt initalization file for starting
  DDR at 800MHz on BootROM stage.
- Add system clock initilization to DDR initalization file
  for ensuring all system clocks and dividers are set correctly
  prior to setting up the DDR controller.
- Beatufy the ddr-600.txt initalization file - add comments,
  extend all values to 4 bytes and convert lowercase HEX
  numbers to uppercase.

Change-Id: I3432b918755cf0f4c56728aa90524e36f7ab9fe4
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27182
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 tools/wtp/ddr-600.txt | 154 ++++++++++++++++++++++++++------------------------
 tools/wtp/ddr-800.txt | 149 ++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 228 insertions(+), 75 deletions(-)
 create mode 100644 tools/wtp/ddr-800.txt

diff --git a/tools/wtp/ddr-600.txt b/tools/wtp/ddr-600.txt
index bba341b..041a960 100644
--- a/tools/wtp/ddr-600.txt
+++ b/tools/wtp/ddr-600.txt
@@ -1,87 +1,91 @@
+; Set DDR to 600MHz
+; Pre-init
 WRITE: 0xC0014008 0x00404500
 WRITE: 0xC0002000 0x00010000
 WRITE: 0xC0002004 0x00000000
-WRITE: 0xC0000340 0x0f0f0fef
-WRITE: 0xC0000344 0x100000aa
-WRITE: 0xC0000310 0x200000
-WRITE: 0xC0000304 0x0
-WRITE: 0xC0000308 0x0
-WRITE: 0xC0000200 0xe0001
-WRITE: 0xC0000204 0x0
+; DDR MAC init
+WRITE: 0xC0000340 0x0F0F0FEF
+WRITE: 0xC0000344 0x100000AA
+WRITE: 0xC0000310 0x00200000
+WRITE: 0xC0000304 0x00000000
+WRITE: 0xC0000308 0x00000000
+WRITE: 0xC0000200 0x000E0001
+WRITE: 0xC0000204 0x00000000
 WRITE: 0xC0000220 0x13020532
-WRITE: 0xC0000044 0x30200
-WRITE: 0xC00002c0 0x6000
+WRITE: 0xC0000044 0x00030200
+WRITE: 0xC00002c0 0x00006000
 WRITE: 0xC00002c4 0x00100020
-WRITE: 0xC0000058 0x143f
-WRITE: 0xC0000048 0x1
+WRITE: 0xC0000058 0x0000143F
+WRITE: 0xC0000048 0x00000001
 WRITE: 0xC0000180 0x00010200
-WRITE: 0xC0000050 0x1ff
-WRITE: 0xC000004c 0x0
-WRITE: 0xC0000054 0x480
-WRITE: 0xC0000300 0x708
-WRITE: 0xC0000380 0x1f5
-WRITE: 0xC0000384 0x3e9
-WRITE: 0xC0000388 0x9600043
-WRITE: 0xC000038c 0x200
-WRITE: 0xC0000390 0x400100
-WRITE: 0xC0000394 0x6b03cf
-WRITE: 0xC0000398 0x720200
-WRITE: 0xC000039c 0x120707
-WRITE: 0xC00003a0 0x40511
-WRITE: 0xC00003a4 0x1
-WRITE: 0xC00003a8 0xc04
+WRITE: 0xC0000050 0x000001FF
+WRITE: 0xC000004c 0x00000000
+WRITE: 0xC0000054 0x00000480
+WRITE: 0xC0000300 0x00000708
+WRITE: 0xC0000380 0x000001F5
+WRITE: 0xC0000384 0x000003E9
+WRITE: 0xC0000388 0x09600043
+WRITE: 0xC000038c 0x00000200
+WRITE: 0xC0000390 0x00400100
+WRITE: 0xC0000394 0x006B03CF
+WRITE: 0xC0000398 0x00720200
+WRITE: 0xC000039c 0x00120707
+WRITE: 0xC00003a0 0x00040511
+WRITE: 0xC00003a4 0x00000001
+WRITE: 0xC00003a8 0x00000C04
 WRITE: 0xC00003ac 0x15210919
-WRITE: 0xC00003b0 0x90b0609
-WRITE: 0xC00003b4 0x4000600
-WRITE: 0xC00003b8 0x600
-WRITE: 0xC00003bc 0x2020404
-WRITE: 0xC00003c0 0x0
-WRITE: 0xC00003c4 0x0
-WRITE: 0xC00003dc 0x81239
-WRITE: 0xC00002c8 0x0
-WRITE: 0xC0000064 0x0006
+WRITE: 0xC00003b0 0x090b0609
+WRITE: 0xC00003b4 0x04000600
+WRITE: 0xC00003b8 0x00000600
+WRITE: 0xC00003bc 0x02020404
+WRITE: 0xC00003c0 0x00000000
+WRITE: 0xC00003c4 0x00000000
+WRITE: 0xC00003dc 0x00081239
+WRITE: 0xC00002c8 0x00000000
+WRITE: 0xC0000064 0x00000006
+; DDR PHY init
 WRITE: 0xC0001004 0x10077779
-WRITE: 0xC0001008 0x1ff00770
-WRITE: 0xC000100c 0x3f03fc77
-WRITE: 0xC0001010 0x100118
-WRITE: 0xC0001028 0x0
-WRITE: 0xC0001030 0x3800000
-WRITE: 0xC0001034 0x0
-WRITE: 0xC0001040 0x400
+WRITE: 0xC0001008 0x1FF00770
+WRITE: 0xC000100c 0x3F03FC77
+WRITE: 0xC0001010 0x00100118
+WRITE: 0xC0001028 0x00000000
+WRITE: 0xC0001030 0x03800000
+WRITE: 0xC0001034 0x00000000
+WRITE: 0xC0001040 0x00000400
 WRITE: 0xC00010c0 0x80000001
-WRITE: 0xC00010d0 0x0
-WRITE: 0xC00010e0 0x11ff0
-WRITE: 0xC0001090 0x0
-WRITE: 0xC0001094 0x0
-WRITE: 0xC0001098 0x0
-WRITE: 0xC000109c 0x0
-WRITE: 0xC00010a0 0x0
-WRITE: 0xC00010a4 0x0
-WRITE: 0xC00010a8 0x0
-WRITE: 0xC00010ac 0x0
-WRITE: 0xC00010b0 0x0
-WRITE: 0xC0001000 0x44041
-WRITE: 0xC0001014 0x80200
-WRITE: 0xC0001038 0x2
-WRITE: 0xC000103c 0x10
-WRITE: 0xC0001180 0x20a
-WRITE: 0xC0001184 0x20a
-WRITE: 0xC0001188 0x20a
-WRITE: 0xC000118c 0x20a
-WRITE: 0xC0001190 0x20a
-WRITE: 0xC0001194 0x20a
-WRITE: 0xC0001198 0x20a
-WRITE: 0xC000119c 0x20a
-WRITE: 0xC00011a0 0x20a
-WRITE: 0xC0001050 0x8080000
-WRITE: 0xC0001054 0x8080000
-WRITE: 0xC0001058 0x8080000
-WRITE: 0xC000105c 0x8080000
-WRITE: 0xC0001060 0x8080000
-WRITE: 0xC0001064 0x8080000
-WRITE: 0xC0001068 0x8080000
-WRITE: 0xC000106c 0x8080000
-WRITE: 0xC0001070 0x8080000
+WRITE: 0xC00010d0 0x00000000
+WRITE: 0xC00010e0 0x00011FF0
+WRITE: 0xC0001090 0x00000000
+WRITE: 0xC0001094 0x00000000
+WRITE: 0xC0001098 0x00000000
+WRITE: 0xC000109c 0x00000000
+WRITE: 0xC00010a0 0x00000000
+WRITE: 0xC00010a4 0x00000000
+WRITE: 0xC00010a8 0x00000000
+WRITE: 0xC00010ac 0x00000000
+WRITE: 0xC00010b0 0x00000000
+WRITE: 0xC0001000 0x00044041
+WRITE: 0xC0001014 0x00080200
+WRITE: 0xC0001038 0x00000002
+WRITE: 0xC000103c 0x00000010
+WRITE: 0xC0001180 0x0000020A
+WRITE: 0xC0001184 0x0000020A
+WRITE: 0xC0001188 0x0000020A
+WRITE: 0xC000118c 0x0000020A
+WRITE: 0xC0001190 0x0000020A
+WRITE: 0xC0001194 0x0000020A
+WRITE: 0xC0001198 0x0000020A
+WRITE: 0xC000119c 0x0000020A
+WRITE: 0xC00011a0 0x0000020A
+WRITE: 0xC0001050 0x08080000
+WRITE: 0xC0001054 0x08080000
+WRITE: 0xC0001058 0x08080000
+WRITE: 0xC000105c 0x08080000
+WRITE: 0xC0001060 0x08080000
+WRITE: 0xC0001064 0x08080000
+WRITE: 0xC0001068 0x08080000
+WRITE: 0xC000106c 0x08080000
+WRITE: 0xC0001070 0x08080000
 WRITE: 0xC0001020 0x20000000
 WRITE: 0xC0001020 0x40000000
 WRITE: 0xC0001020 0x80000000
diff --git a/tools/wtp/ddr-800.txt b/tools/wtp/ddr-800.txt
new file mode 100644
index 0000000..210b2c8
--- /dev/null
+++ b/tools/wtp/ddr-800.txt
@@ -0,0 +1,149 @@
+; Switch all clocks to REFCLOCK
+WRITE: 0XC0013010 0x00000000
+WRITE: 0XC0018010 0x00000000
+; TBG-A: SE vco_div 0x0,
+; DIFF vco_div 0x1, vco_range 0xa
+; tbg_N 0x30 KVCO = 1600 MHz
+WRITE: 0xC0013204 0x00C00091
+WRITE: 0xC0013204 0x00C000C1
+WRITE: 0xC0013220 0x08030803
+WRITE: 0xC0013208 0x94011400
+WRITE: 0xC0013230 0x00020002
+WRITE: 0xC0013208 0x94011400
+WRITE: 0xC001320C 0x53E556E6
+WRITE: 0xC0013210 0x014A014A
+WRITE: 0xC001320C 0x53E556E5
+WRITE: 0xC0013204 0x00C000C0
+WRITE: 0xC0013208 0x94011400
+WAIT_FOR_BIT_SET: 0xC0013208 0x80008000 0x00100000
+DELAY: 0x00000100
+; TBG-B: SE vco_div 0x1,
+; DIFF vco_div 0x2, vco_range 0xb
+; tbg_N 0x3c KVCO = 2000 MHz
+WRITE: 0xC0013204 0x00C100C0
+WRITE: 0xC0013204 0x00F100C0
+WRITE: 0xC0013220 0x08030803
+WRITE: 0xC0013208 0x14019400
+WRITE: 0xC0013230 0x00040002
+WRITE: 0xC0013208 0x14019400
+WRITE: 0xC001320C 0x56E556E5
+WRITE: 0xC0013210 0x014B014A
+WRITE: 0xC001320C 0x56E656E5
+WRITE: 0xC0013204 0x00F000C0
+WRITE: 0xC0013208 0x14019400
+WAIT_FOR_BIT_SET: 0xC0013208 0x80008000 0x00100000
+DELAY: 0x00000100
+; Set clocks to 800/800 preset
+WRITE: 0xC0013014 0x00000000
+WRITE: 0xC0013004 0x1296202C
+WRITE: 0xC0013008 0x21061AA9
+WRITE: 0xC001300C 0x20543082
+WRITE: 0xC0013000 0x0003C0F2
+WRITE: 0xC0018014 0x00180000
+WRITE: 0xC0018004 0x02515508
+WRITE: 0xC0018008 0x00307880
+WRITE: 0xC001800C 0x00000540
+WRITE: 0xC0018000 0x003F8F40
+WRITE: 0xC0013210 0x0014B014A
+; Switch all clocks to back dividers
+WRITE: 0xC0013010 0x00009FFF
+WRITE: 0xC0018010 0x000007AA
+; Set DDR to 800MHz
+; Pre-init
+WRITE: 0xC0014008 0x00404500
+WRITE: 0xC0002000 0x00010000
+WRITE: 0xC0002004 0x00000000
+; DDR MAC init
+WRITE: 0xC0000340 0x0F0F0FEF
+WRITE: 0xC0000344 0x100000AA
+WRITE: 0xC0000310 0x00200000
+WRITE: 0xC0000304 0x00000000
+WRITE: 0xC0000308 0x00000000
+WRITE: 0xC0000200 0x000e0001
+WRITE: 0xC0000204 0x00000000
+WRITE: 0xC0000220 0x13020532
+WRITE: 0xC0000044 0x00030200
+WRITE: 0xC00002c0 0x00006000
+WRITE: 0xC00002c4 0x00100020
+WRITE: 0xC0000058 0x0000143F
+WRITE: 0xC0000048 0x00000001
+WRITE: 0xC0000180 0x00010200
+WRITE: 0xC0000050 0x000001FF
+WRITE: 0xC000004c 0x00000000
+WRITE: 0xC0000054 0x00000480
+WRITE: 0xC0000300 0x0000080B
+WRITE: 0xC0000380 0x0007A120
+WRITE: 0xC0000384 0x00030D40
+WRITE: 0xC0000388 0x0960006B
+WRITE: 0xC000038c 0x00000200
+WRITE: 0xC0000390 0x00400100
+WRITE: 0xC0000394 0x00F003CF
+WRITE: 0xC0000398 0x00F80200
+WRITE: 0xC000039c 0x00000808
+WRITE: 0xC00003a0 0x00040614
+WRITE: 0xC00003a4 0x00000001
+WRITE: 0xC00003a8 0x00000C04
+WRITE: 0xC00003ac 0x202A0C1F
+WRITE: 0xC00003b0 0x0C0C060C
+WRITE: 0xC00003b4 0x04000600
+WRITE: 0xC00003b8 0x00000800
+WRITE: 0xC00003bc 0x02020404
+WRITE: 0xC00003c0 0x00000000
+WRITE: 0xC00003c4 0x00000000
+WRITE: 0xC00003dc 0x00081239
+WRITE: 0xC00002c8 0x00000000
+WRITE: 0xC0000064 0x00000006
+; DDR PHY init
+WRITE: 0xC0001004 0x10077779
+WRITE: 0xC0001008 0x1FF00770
+WRITE: 0xC000100c 0x3F03FC77
+WRITE: 0xC0001010 0x00100118
+WRITE: 0xC0001028 0x00000000
+WRITE: 0xC0001030 0x03800000
+WRITE: 0xC0001034 0x00000000
+WRITE: 0xC0001040 0x00000400
+WRITE: 0xC00010c0 0x80000001
+WRITE: 0xC00010d0 0x00000000
+WRITE: 0xC00010e0 0x00011FF0
+WRITE: 0xC0001090 0x00000000
+WRITE: 0xC0001094 0x00000000
+WRITE: 0xC0001098 0x00000000
+WRITE: 0xC000109c 0x00000000
+WRITE: 0xC00010a0 0x00000000
+WRITE: 0xC00010a4 0x00000000
+WRITE: 0xC00010a8 0x00000000
+WRITE: 0xC00010ac 0x00000000
+WRITE: 0xC00010b0 0x00000000
+WRITE: 0xC0001000 0x00044041
+WRITE: 0xC0001014 0x00080200
+WRITE: 0xC0001038 0x00000002
+WRITE: 0xC000103c 0x00000010
+WRITE: 0xC0001180 0x0000020A
+WRITE: 0xC0001184 0x0000020A
+WRITE: 0xC0001188 0x0000020A
+WRITE: 0xC000118c 0x0000020A
+WRITE: 0xC0001190 0x0000020A
+WRITE: 0xC0001194 0x0000020A
+WRITE: 0xC0001198 0x0000020A
+WRITE: 0xC000119c 0x0000020A
+WRITE: 0xC00011a0 0x0000020A
+WRITE: 0xC0001050 0x20200000
+WRITE: 0xC0001054 0x20200000
+WRITE: 0xC0001074 0x20200000
+WRITE: 0xC0001058 0x08080000
+WRITE: 0xC000105c 0x08080000
+WRITE: 0xC0001060 0x08080000
+WRITE: 0xC0001064 0x08080000
+WRITE: 0xC0001068 0x08080000
+WRITE: 0xC000106c 0x08080000
+WRITE: 0xC0001070 0x08080000
+WRITE: 0xC0001020 0x20000000
+WRITE: 0xC0001020 0x40000000
+WRITE: 0xC0001020 0x80000000
+; Trigger DDR init for channel 0
+WRITE: 0xC0000020 0x11000001
+WAIT_FOR_BIT_SET: 0xC0000008 0x00000001 0x00001000
+; AVS enable
+WRITE: 0xC0011500 0x98A2FFFF	;# Set VDD to 1150mV and soft reset AVS
+WRITE: 0xC0011500 0x18A2FFFF	;# release AVS reset
+WRITE: 0xC0011500 0x58A2FFFF	;# Enable AVS
-- 
1.9.1

